File: packages/device_calendar_plus_android/android/src/main/kotlin/to/bullet/device_calendar_plus_android/ColorHelper.kt
package to.bullet.device_calendar_plus_android

import android.graphics.Color

object ColorHelper {
    /** [hex] is the canonical `#RRGGBB` Dart's `normalizeColorHex` forwards. */
    fun hexToColor(hex: String): Int = Color.parseColor(hex)

    fun colorToHex(color: Int): String {
        // Android color is ARGB, we want RGB hex string
        return String.format("#%06X", 0xFFFFFF and color)
    }
}

File: packages/device_calendar_plus_ios/ios/device_calendar_plus_ios/Sources/device_calendar_plus_ios/ColorHelper.swift
import Foundation
import CoreGraphics

class ColorHelper {
  /// `hex` is the canonical `#RRGGBB` Dart's `normalizeColorHex` forwards.
  static func hexToColor(hex: String) -> CGColor {
    let rgb = UInt64(hex.dropFirst(), radix: 16) ?? 0

    let r = CGFloat((rgb & 0xFF0000) >> 16) / 255.0
    let g = CGFloat((rgb & 0x00FF00) >> 8) / 255.0
    let b = CGFloat(rgb & 0x0000FF) / 255.0
    
    return CGColor(red: r, green: g, blue: b, alpha: 1.0)
  }
  
  static func colorToHex(cgColor: CGColor) -> String {
    guard let components = cgColor.components, components.count >= 3 else {
      return "#000000"
    }
    
    let r = Int(components[0] * 255.0)
    let g = Int(components[1] * 255.0)
    let b = Int(components[2] * 255.0)
    
    return String(format: "#%02X%02X%02X", r, g, b)
  }
}

